src/components/TopBar/AboutDialog.js

```diff
@@ -58,39 +150,46 @@ function compareModules(c1, c2) {
 const AboutDialog = ({
     open,
     onClose,
-    getGlobalVersion,
+    globalVersionPromise,
     appName,
     appVersion,
     appGitTag,
     appLicense,
-    getAdditionalModules,
+    additionalModulesPromise,
 }) => {
     const theme = useTheme();
     const [isRefreshing, setRefreshState] = useState(false);
     const [loadingGlobalVersion, setLoadingGlobalVersion] = useState(false);
+    const [showGlobalVersion, setShowGlobalVersion] = useState(false);
 
     /* We want to get the initial version once at first render to detect later a new deploy */
     const [startingGlobalVersion, setStartingGlobalVersion] =
         useState(undefined);
     useEffect(() => {
-        if (startingGlobalVersion === undefined && getGlobalVersion) {
-            getGlobalVersion((value) => {
-                setStartingGlobalVersion(value);
-                setActualGlobalVersion(value);
-            });
+        if (startingGlobalVersion === undefined && globalVersionPromise) {
+            getGlobalVersion(
+                globalVersionPromise,
+                'Initial',
+                setStartingGlobalVersion,
+                undefined
+            );
         }
-    }, [getGlobalVersion, startingGlobalVersion]);
+    }, [globalVersionPromise, startingGlobalVersion]);
 
     const [actualGlobalVersion, setActualGlobalVersion] = useState(null);
     useEffect(() => {
-        if (open && getGlobalVersion) {
-            setLoadingGlobalVersion(true);
-            getGlobalVersion((value) => {
-                setLoadingGlobalVersion(false);
-                setActualGlobalVersion(value || null);
-            });
+        if (open && globalVersionPromise) {
+            getGlobalVersion(
+                globalVersionPromise,
+                'Actual',
+                setActualGlobalVersion,
+                (value) => {
+                    setLoadingGlobalVersion(value);
```

Comment:
```suggestion
                    setLoadingGlobalVersion(!!value);
```
More comprehensive no ?